cfr_tool/explosives.py

```diff
@@ -0,0 +1,136 @@
+import re
+import clean_text as ct
+
+class Explosives:
+    def __init__(self, db, soup):
+        self.db = db
+        self.soup = soup
+
+    def create_load_explosives(self):
+        self.parse_load_pis()
+        self.parse_load_packing_methods()
+    
+    def parse_load_pis(self):
+        self.db.executescript("DROP TABLE IF EXISTS explosives_table;")
+        self.db.executescript('''
+            CREATE TABLE explosives_table (
+                id_num text,
+                packaging_instruction text,
+                FOREIGN KEY (id_num)
+                    REFERENCES hazmat_table (id_num)
+            )
+        ''')
+        explosives = self.soup.find_table("Explosives Table")
+
+        entries = []
+        for row in explosives.find_all('row'):
+            text = [ent.text for ent in row.find_all('ent')]
+            un_na = text[0]
+            pis = text[1].split(' or ')
+            for pi in pis:
+                entries.append((un_na, pi))
+        self.db.executemany('''
+            INSERT INTO explosives_table (id_num, packaging_instruction)
+            VALUES (?, ?)
+        ''', entries)
+
+    def create_packing_methods(self):
+        self.db.executescript("DROP TABLE IF EXISTS packing_methods;")
+        self.db.executescript('''
+            CREATE TABLE packing_methods (
+                packaging_instruction text,
+                inner_packagings_type text,
+                intermediate_packagings_type text,
+                outer_packagings_type text,
+                requirements_exceptions text,
+                inner_packagings_material text,
+                intermediate_packagings_material text,
+                outer_packagings_material text,
+                FOREIGN KEY (packaging_instruction)
+                    REFERENCES explosives_table (packaging_instruction)
+            );
+        ''')
+        pass
+
+    def create_outer_packagings(self):
+        self.db.executescript("DROP TABLE IF EXISTS outer_packagings;")
+        self.db.executescript('''
+            CREATE TABLE outer_packagings (
+                packaging_instruction text,
+                packaging_code text,
+                FOREIGN KEY (packaging_instruction)
+                    REFERENCES explosives_table (packaging_instruction),
+                FOREIGN KEY (packaging_code)
+                    REFERENCES packagings (packaging_code)
+            );
+        ''')
```

Comment:
I wonder if some of this SQL will become more general (e.g. outer packaging table would incorporate all the info we get from each section/subsection of the CFR that we parse through)